lab-justin/server.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const http = require('http');
+const port = process.env.PORT || 3000;
+const parseUrl = require('url').parse;
+const parseQuery = require('querystring').parse;
+const bodyParser = require('./lib/bodyParser');
+const cowsay = require('cowsay');
+
+const server = http.createServer(function(req, res){
+  req.url = parseUrl(req.url);
+  console.log('req.url', req.url);
+  req.url.query = parseQuery(req.url.query);
+  console.log('req.method', req.method);
+
+  if (req.url.pathname === '/'){
+    res.writeHead(200, {'Content-Type': 'text/plain'});
+    res.write(JSON.stringify('API Endpoints: \n /api/cowsay'));
+    return res.end();
+  }
+
+  if(req.method ==='POST' && req.url.pathname ==='/api/cowsay'){
+    bodyParser(req).then(function(){
+
```

Comment:
The way this works, I think they could submit an empty string and have the cow 'say' it, but the assignment didn't specify what to do in that case, so this is totally valid.